syft/pkg/terraform.go

```diff
@@ -0,0 +1,9 @@
+package pkg
+
+// TerraformLockEntry represents a single entry in a Terraform dependency lock file (.terraform.lock.hcl).
+type TerraformLockEntry struct {
```

Comment:
is this intended to exclusively represent providers? or can it also describe modules? From the cataloger it seems like this would always be a provider, if so this should probably be `TerraformProviderLockEntry` or similar

Reply:
or another way to put it: does this lock file describe only providers? or can it describe other things too?

Reply:
The lock file describes two different kinds of external dependencies:
- providers
- modules

See https://developer.hashicorp.com/terraform/language/files/dependency-lock for details. However the site also states:
> At present, the dependency lock file tracks only provider dependencies.

Reply:
Gotcha, after some thought, I think this should be `TerraformLockProviderEntry` since there could be more entities to describe in the future and we want to be able to distinct them. This is also more accurate and helps the end user better understand what is being cataloged.